lib/jake/array.ex

```diff
@@ -1,5 +1,99 @@
 defmodule Jake.Array do
-  def gen(_) do
-    StreamData.constant([])
+  @type_list [
+    %{"type" => "integer"},
+    %{"type" => "number"},
+    %{"type" => "boolean"},
+    %{"type" => "string"},
+    %{"type" => "null"},
+    nil
+  ]
+
+  @min_items 0
+
+  @max_items 1000
+
+  def gen_array(map, type), do: arraytype(map, map["enum"], map["items"])
+
+  def arraytype(map, enum, items) when enum != nil, do: Jake.gen_enum(enum, "array")
+
+  def arraytype(map, enum, items) when is_list(items) or is_map(items) do
+    list =
+      if is_list(items) do
+        for n <- items, is_map(n), do: Jake.gen_init(n)
+      else
+        [Jake.gen_init(items)]
+      end
+
+    {min, max} = get_min_max(map)
+
+    case map["additionalItems"] do
+      x when (is_boolean(x) and x) or is_nil(x) ->
+        add_additional_items(list, true, max, min)
+
+      x when is_map(x) ->
+        add_additional_items(list, x, max, min)
+
+      _ ->
+        add_additional_items(list, false, max, min)
+    end
+  end
+
+  def get_min_max(map) do
+    min = Map.get(map, "minItems", @min_items)
+    max = Map.get(map, "maxItems", @max_items)
+    {min, max}
+  end
+
+  def arraytype(map, enum, items) when is_nil(items) and is_nil(enum) do
+    item = get_one_of()
+    {min, max} = get_min_max(map)
+    decide_min_max(map, item, min, max)
+  end
+
+  def decide_min_max(map, item, min, max)
+      when is_integer(min) and is_integer(max) and min < max do
+    if map["uniqueItems"] do
+      StreamData.uniq_list_of(item, min_length: min, max_length: max)
+    else
+      StreamData.list_of(item, min_length: min, max_length: max)
+    end
+  end
+
+  def get_one_of() do
+    for(n <- @type_list, is_map(n), do: Jake.gen_init(n)) |> StreamData.one_of()
```

Comment:
could this reuse the notype generator

Reply:
In notype, a type will be added to the schema depending on the properties. 

In get_one_of, it is not needed, since we already know type, here we using it to generate random values from random types, when no information is given about array. I think here, we can also use ``StreamData.term()`` . What do you think?

Reply:
`StreamData.term()` might not generate valid json values, tuple for example. I was assuming notype should handle cases like empty spec `{}`, and would generate some valid value